### PR TITLE
FACES-2816 Develop tests for the Button/Link category of the Liferay …

### PR DESCRIPTION
**Pull request by pcwhite**
…Faces Showcase (additionally improved Link General use case consistency)

### CODE REVIEW

**Comment by pcwhite** [comment]
besides webapp, also tested on liferay 7 and pluto 3.

File: demo/alloy-showcase-webapp/src/main/webapp/WEB-INF/component/alloy/button/menu/button.xhtml [hunk 1]
```diff
@@ -56,7 +56,7 @@
 					<alloy:button includeViewParams="true">
 						<alloy:icon name="user" />
 						<alloy:outputText value="#{i18n['user']}" />
-						<f:param name="menuItem" value="view" />
```
Comment (pcwhite) [inline demo/alloy-showcase-webapp/src/main/webapp/WEB-INF/component/alloy/button/menu/button.xhtml:59]: for consistency sake and better testing coverage, I altered the outputText value from "view" to "user" (commandMenu says you selected 'User' in feedback text after clicking on the User button).

File: demo/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/alloy/showcase/buttonlink/ButtonLinkTester.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		testMenuItemSelection(browser, exampleType, dropdownType, menuItemOutputXpath, "dit");
+		testMenuItemSelection(browser, exampleType, dropdownType, menuItemOutputXpath, "nable");
+		testMenuItemSelection(browser, exampleType, dropdownType, menuItemOutputXpath, "isable");
+		testMenuItemSelection(browser, exampleType, dropdownType, menuItemOutputXpath, "elete");
+	}
+
+	protected void testMenuItemSelection(Browser browser, String exampleType, String dropdownType,
+		String menuItemOutputXpath, String menuItem) {
+		String dropdownButtonXpath = "//span[@role='menuitem']/a[contains(@class,'btn btn-default alloy-')]/span[contains(@class,'" + dropdownType +
+				"')]";
```
Comment (pcwhite) [inline demo/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/alloy/showcase/buttonlink/ButtonLinkTester.java]: Updated!

**Comment by stiemannkj1** [comment]
@pcwhite,
Just finished my code review commit. I have tested it on all browsers in webapps, but I need you to test it in the portals.

Please retest the following permutations with my changes:

- [ ] Liferay 7 (JSF and Alloy Showcases) with Firefox
- [ ] Liferay 7 (JSF and Alloy Showcases) with Chrome
- [ ] Liferay 7 (JSF and Alloy Showcases) with PhantomJS

- [ ] Pluto 2 (JSF and Alloy Showcases) with Firefox
- [ ] Pluto 2 (JSF and Alloy Showcases) with Chrome
- [ ] Pluto 2 (JSF and Alloy Showcases) with PhantomJS

In order to test my changes, you will need to follow these steps (which may not be exhaustive):

```
# Workaround i18n bug:

cd ~/Projects/liferay.com/liferay-faces-util-3.x

git checkout d2eb96a6f6ef95a7a80507899a848c07f230cf64

mvn clean install

# Install my changes to the showcase:

cd ~/Projects/liferay.com/liferay-faces-showcase-3.x

git fetch stiemannkj1

git checkout pcwhite_FACES-2816-3.x

mvn clean install

# Install my changes to alloy

cd ~/Projects/liferay.com/liferay-faces-alloy

git fetch stiemannkj1

git checkout pcwhite_FACES-2816

# Test only the showcases

cd ~/Projects/liferay.com/liferay-faces-bridge-4.x/test/integration

mvn verify -P selenium,$BROWSER_NAME,$PORTAL_NAME -Dit.test=*.showcase.*.*Tester
```

**Comment by stiemannkj1** [comment]
@pcwhite,
I forgot to mention this: before running the full showcase test suite, please test **only** the new components **once** (on chrome) to make sure that you did everything correctly. You can use this command:

    mvn verify -P selenium,chrome,pluto -Dit.test=%regex[.*(Button|Link|Image).*]

Just make sure you confirm (visually) that new tests like the `alloy:commandButton` *Menu* use-case test are being run.

Thanks.

**Comment by stiemannkj1** [comment]
@pcwhite, merged and sent to Neil as https://github.com/liferay/liferay-faces-alloy/pull/121. See code review commit here: https://github.com/liferay/liferay-faces-alloy/pull/121/commits/754ab16d1b7e8118aa652790e3084ed6d12dbea8

Please make sure that you test every permutation in the matrix (including Alloy) before sending a PR.

**Pull request closed without merging**